e2e_tests/tests/command/test_shell.py

```diff
@@ -12,10 +12,16 @@ def test_start_and_write_to_shell(tmp_path: Path) -> None:
     with cmd.interactive_command("shell", "start") as shell:
         # Call our cli to ensure that PATH and PYTHONUSERBASE are properly set.
         shell.stdin.write(b"COLUMNS=80 det --version\n")
+        # Exit the shell, so we can read output below until EOF instead of timeout
+        shell.stdin.write(b"exit\n")
         shell.stdin.close()
 
+        lines = ""
         for line in shell.stdout:
             if str(det.__version__) in line:
                 break
+            lines += "OUTPUT:" + line
```

Comment:
"line" hasn't had the newline character stripped out?

Reply:
> "line" hasn't had the newline character stripped out?
Nope.  I added it and it was all double spaced and I had to take it out.  See failure message in PR description.